utilities/translation/clean.js

```diff
@@ -0,0 +1,185 @@
+    // transforming to lower case every key and removing unuseful spaces
+    key = key.toLowerCase().trim();
+
+    // Evoiding dealing with value parsed in angular format and other
+    if (key.indexOf('{') === -1 &&
+        key.indexOf('}') === -1 &&
+        key.indexOf('$') === -1 &&
+        key !== '') {
+        try {
+            console.log('Processing', key, '.json', '...');
+            var jsonRoute = path.resolve(__dirname, i18nRoute, key + '.json');
+
+            //loading the small json file used for developpement
+            let obj = require(jsonRoute);
+
+            // for every i18n json file, this function will return an array of keyString with the format : KEY1.KEY2.KEY3
+            var list = getAllKeyString(obj, '', []);
+
+            //processing every single key string
+            list.forEach((item) => {
+                // return true if the key string is present in the object containing data to remove (en.json)
+                if (isPresent(item)) {
+
+                    //if true, the last key is removed from the small json file obj
+                    obj[item.split('.')[0]] = removeLastKey(item, obj);
+                }
+            });
+
+            // stringify the object in order to write with utf8 encoding
+            const objAsString = JSON.stringify(obj);
+            const f = fs.openSync(jsonRoute, 'w');
+
+            // Writting  a new content into the small json file i18n
+            fs.writeFileSync(f, objAsString, 'utf8');
+        }
+        catch (e) {
+            //FIX ME : in case require can not find the file, how can I handle it properly?
+        }
+    }
+});
+
+function getAllKeyString(obj, path, res) {
+    res = res || [];
+
+    keys = Object.keys(obj).sort();
+
+    keys.forEach((key) => {
+        if (typeof obj[key] === 'object') {
+            if (path.length > 0) {
+                return res.concat(getAllKeyString(obj[key], path + '.' + key, res));
+            } else {
+                return res.concat(getAllKeyString(obj[key], key, res));
+            }
+        } else {
+            res.push(path.length > 0 ? path + '.' + key : key);
+        }
+
+    });
+
+    return res
+}
+
+function getTemplates(route, res) {
+    res = res || [];
+
+    if (fs.lstatSync(route).isDirectory()) {
+        fs.readdirSync(route).forEach((file) => {
+            return res.concat(getTemplates(path.resolve(route, file), res));
+        });
+    } else if (!fs.lstatSync(route).isDirectory() && path.extname(route) === selectedExtension) {
+        res.push(route);
+    }
+
+    return res;
+}
+
+function removeLastKey(keyString, baseObejct) {
+    let keys;
+    keys = keyString.split('.');
+
+    return keys.reduce((current, next) => {
+        if (baseObejct[next]) {
+            return current[next] = baseObejct[next];
+        } else {
+            if (typeof current[next] === 'string') {
+                delete current[next];
+                return current;
```

Comment:
You can move this `return current` out of the `if` for simplicity.  It doesn't depend on the `if` logic.
```js
if (typeof current[next] === 'string') {
  delete current[next];
}
return current;
```